src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -433,6 +471,19 @@ private static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAcce
             return instance ?? throw new InvalidOperationException(SR.Format(SR.Error_FailedToActivate, type));
         }
 
+        private static bool CanBindToTheseConstructorParameters(ParameterInfo[] constructorParameters)
```

Comment:
This could be greatly simplified using LINQ

Reply:
Yes, I agree, although there is a [push to reduce the use of Linq for less allocations.](https://github.com/dotnet/runtime/issues/44598)